client/allocrunner/taskrunner/template/template.go

```diff
@@ -525,6 +525,20 @@ func templateRunner(config *TaskTemplateManagerConfig) (
 	return runner, lookup, nil
 }
 
+// maskProcessEnv masks away any environment variable not found in task env.
+// It manipulates the parameter directly and returns it without copying.
+func maskProcessEnv(env map[string]string) map[string]string {
```

Comment:
config.EnvBuilder should already be a safe subset of environment variables to expose to tasks. If there's a reason to restrict the set of environment variables to consul-template even further, perhaps we can ad that functionality directly to the `taskenv` package?

This seems like a potentially large backward compat change for something that should already work as desired.

Reply:
@notnoop pointed out the code linked in the description to me: consul-template itself falls back to host env vars, so we have to mask them manually.